src/rest_api/amoc_api_logic_handler.erl

```diff
@@ -0,0 +1,84 @@
+%%==============================================================================
+%% Copyright 2015 Erlang Solutions Ltd.
+%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
+%%==============================================================================
+-module(amoc_api_logic_handler).
+
+-behaviour(amoc_rest_logic_handler).
+
+-export([handle_request/3]).
+
+-spec handle_request(
+    OperationID :: amoc_rest_api:operation_id(),
+    Req :: cowboy_req:req(),
+    Context :: #{}
+) ->
+    {Status :: cowboy:http_status(), Headers :: cowboy:http_headers(), Body :: jsx:json_term()}.
+
+handle_request('NodesGet', _Req, _Context) ->
+    Status = amoc_cluster:get_status(),
+    Connected = maps:get(connected, Status, []),
+    FailedToConnect = maps:get(failed_to_connect, Status, []),
+    ConnectionLost = maps:get(connection_lost, Status, []),
+    Up = [{Node, <<"up">>} || Node <- [node() | Connected]],
+    DownNodes = lists:usort(FailedToConnect ++ ConnectionLost),
+    Down = [{Node, <<"down">>} || Node <- DownNodes],
+    ResponseList = Up ++ Down,
+    {200, #{}, [{<<"nodes">>, ResponseList}]};
+handle_request('ScenariosGet', _Req, _Context) ->
+    Scenarios = amoc_scenario:list_scenario_modules(),
+    BinaryScenarios = [atom_to_binary(S, latin1) || S <- Scenarios],
```

Comment:
Is there a reason behind the `latin1` encoding in `amoc`? Maybe `utf8` could be used here?

Reply:
latin1 should be good enough for the module name, but I can change it to utf8 if you want.